examples/src/python/example/python_distribution/hello/superhello/BUILD

```diff
@@ -0,0 +1,16 @@
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# Like Hello world, but built with a python_distribution.
+# python_distribution allows you to use setup.py to depend on C/C++ extensions.
+
+# There must be a single BUILD file and a setup.py in a given python_distribution.
+
+# All you need to indicate is a name and any dependencies.
+# You are expected to define source locations in setup.py.
+# There should be a one-to-one mapping for the name of the python_distribution
+# and the name of the python_distribution's directory. In this case, it's superhello.
+
+python_distribution(
+  name='superhello'
+)
```

Comment:
Along the lines of #3, what if we just used rglobs(*.py|*.c|*.cpp) as an implicit default? This way the user would not need to define the sources they care about in two places (setup.py and then have to also list them out in a sources param). I'm not sure I understand the value in allowing the user to define sources beyond restricting change detection to a subset of rglobs(*.py|*.c|*.cpp).

Reply:
1) `rglobs` would prevent `python_dist` from being able to co-exist with other targets in a dir or any child dir, which I think would be bad in general. I'm fairly sure (but not positive) that all of the existing implicit sources globs you'd find would be regular `globs` for this reason.
2) globbing all of the .h/.c/.cpp to me, for `python_dist`, feels unusual/unexpected as a user. I could see `*.py` - but I'd argue that C extension dists are typically an exception vs the norm in the general case, so we should probably require that users explicitly declare those when needed.

I'm open to other perspectives and counter arguments tho, but from an adoption strategy perspective in my experience it's way easier to add shortcuts like this later than it would be to take them away once delivered if we find they lead to bad/unwanted behavior - so IMHO, starting with the least surprising implicit behavior would be ideal.

Reply:
✅